specification/dns/resource-manager/Microsoft.Network/stable/2017-10-01/dns.json

```diff
@@ -0,0 +1,1584 @@
+          },
+          "List SOA recordsets": {
+            "$ref": "./examples/ListSOARecordset.json"
+          },
+          "List SRV recordsets": {
+            "$ref": "./examples/ListSRVRecordset.json"
+          },
+          "List TXT recordsets": {
+            "$ref": "./examples/ListTXTRecordset.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/dnsZones/{zoneName}/recordsets": {
+      "get": {
+        "tags": [
+          "RecordSets"
+        ],
+        "operationId": "RecordSets_ListByDnsZone",
+        "description": "Lists all record sets in a DNS zone.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "zoneName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the DNS zone (without a terminating dot)."
+          },
+          {
+            "name": "$top",
+            "in": "query",
+            "required": false,
+            "type": "integer",
+            "format": "int32",
+            "description": "The maximum number of record sets to return. If not specified, returns up to 100 record sets."
+          },
+          {
+            "name": "$recordsetnamesuffix",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "The suffix label of the record set name that has to be used to filter the record set enumerations. If this parameter is specified, Enumeration will return only records that end with .<recordSetNameSuffix>"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success.",
+            "schema": {
+              "$ref": "#/definitions/RecordSetListResult"
+            }
+          },
+          "default": {
+            "description": "Default response. It will be deserialized as per the Error definition.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "List recordsets by zone": {
+            "$ref": "./examples/ListRecordSetsByZone.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/dnsZones/{zoneName}/all": {
+      "get": {
+        "tags": [
+          "RecordSets"
+        ],
+        "operationId": "RecordSets_ListAllByDnsZone",
```

Comment:
>RecordSets_ListAllByDnsZone [](start = 24, length = 27)

Could x-ms-odata extension be useful here?

Reply:
Doesn't look like it. The autorest generator for the C# SDK fails if I add that due to not having $filter parameter, which we do not support in this API.

FATAL: System.InvalidOperationException: Method with x-ms-odata needs to have "$filter" parameter.

Reply:
Okok